### PR TITLE
[2.0] Support SDK for det and cls

### PR DESCRIPTION
**Pull request by AllentDan**
Please note that the results of the SDK classification demo will be different from mmcls as ResizeEdge is replaced to Resize.

### CODE REVIEW

File: mmdeploy/codebase/mmcls/deploy/classification.py [hunk 1]
```diff
@@ -168,18 +168,11 @@ def create_input(
         Returns:
             tuple: (data, img), meta information for the input image and input.
         """
-
+        assert 'test_pipeline' in self.model_cfg, \
```
Comment (lvhan028) [inline mmdeploy/codebase/mmcls/deploy/classification.py:171]: May remove `\` in log

File: mmdeploy/codebase/mmdet/deploy/object_detection.py [hunk 2]
```diff
@@ -214,24 +214,58 @@ def get_preprocess(self, *args, **kwargs) -> Dict:
...
+        transforms.insert(
+            -2,
+            dict(
+                type='Pad',
+                size_divisor=data_preprocessor.get('pad_size_divisor', 1)))
+        transforms.insert(
+            -3,
+            dict(
+                type='Normalize',
+                to_rgb=data_preprocessor.get('bgr_to_rgb', False),
```
Comment (lvhan028) [inline mmdeploy/codebase/mmdet/deploy/object_detection.py:248]: Isn't the default value of `to_rgb` `True`?

Comment (AllentDan) [reply mmdeploy/codebase/mmdet/deploy/object_detection.py:248]: The default value of it in `ImgDataPreprocessor` is False.

File: mmdeploy/codebase/mmcls/deploy/classification.py (hunk 1 above)
Comment (AllentDan) [reply mmdeploy/codebase/mmcls/deploy/classification.py:171]: `\` here is outside the string.

**Review by lvhan028 — APPROVED** [review]

**Review by irexyc — COMMENTED** [review]
感觉新的pipeline中resize的参数有点冗余，建议只保留size。然后关于mmcls中的reisze，之前sdk里面存在"edge": "short"的逻辑（scale, -1) 表示短边缩放到scale.

```
mask-rcnn
新：
{
     "type": "Resize",
     "scale": [
         1344,
         800
     ],
     "keep_ratio": false,
     "size": [
         1344,
         800
     ]
},

旧:
{
    "type": "Resize",
    "keep_ratio": true,
    "size": [
        800,
        1333
    ]
},

resnet18-imagenet
新：
{
    "type": "Resize",
    "keep_ratio": true,
    "size": [
        800,
        1333
    ]
},
旧：
{
    "type": "Resize",
    "size": [
        256,
        -1
    ]
},

**Pull request merged**